core/src/system_state.rs

```diff
@@ -13,10 +13,17 @@ static IS_RUNNING: Lazy<AtomicBool> = Lazy::new(|| AtomicBool::new(true));
 pub async fn initiate_shutdown() {
     IS_RUNNING.store(false, Ordering::SeqCst);
     let mut active = active_indexing_count();
+    let shutdown_timeout = Duration::from_secs(15);
+    let start_shutdown = Instant::now();
 
     info!("Starting shutdown with {} active tasks", active);
 
     loop {
+        if start_shutdown.elapsed() >= shutdown_timeout && active <= 1 {
```

Comment:
Never happens on `rust` project, but sometimes it does seem to happen on nocode projects

Reply:
add a comment in why i think here

Reply:
I'll remove it actually, it's annoying but doesnt' really impact me anymore and it's probably better to just solve it if we start encountring it again.

I think panicking threads is one cause of this, but there may be others.

Reply:
Happy to keep if it solves a real issue just a comment above it explaining would be helpful in the code